net/buffer_pool.c

```diff
@@ -89,8 +89,7 @@ M0_INTERNAL int m0_net_buffer_pool_init(struct m0_net_buffer_pool *pool,
 
 	M0_PRE(pool != NULL);
 	M0_PRE(ndom != NULL);
-	M0_PRE(seg_nr   <= m0_net_domain_get_max_buffer_segments(ndom));
-	M0_PRE(seg_size <= m0_net_domain_get_max_buffer_segment_size(ndom));
```

Comment:
Why is this checking removed?
If this assertion is removed, do we need to handle error case?

Reply:
This function is common for rpc ping and bulk buffer pools and the ping and bulk pools can have different configurations for segment size and seg_nr.
For example : ping buffer pool for libfabric has seg_size = 1MB and seg_nr = 1,
whereas the bulk buffer pool for libfabric has seg_size = 4k and seg_nr = 256